deploy/crds/mongodb.com_mongodb_crd.yaml

```diff
@@ -50,14 +50,95 @@ spec:
+                        or from.
+                      properties:
+                        clientSource:
+                          items:
+                            type: string
+                          type: array
+                        serverAddress:
+                          items:
+                            type: string
+                          type: array
+                      required:
+                      - clientSource
+                      - serverAddress
+                      type: object
+                    type: array
+                  db:
+                    description: The database of the role.
+                    type: string
+                  privileges:
+                    description: The privileges to grant the role.
+                    items:
+                      description: Privilege defines the actions a role is allowed
+                        to perform on a given resource.
+                      properties:
+                        actions:
+                          items:
+                            type: string
+                          type: array
+                        resource:
+                          description: Resource specifies specifies the resources
+                            upon which a privilege permits actions. See https://docs.mongodb.com/manual/reference/resource-document
+                            for more.
+                          properties:
+                            anyResource:
+                              type: boolean
+                            cluster:
+                              type: boolean
+                            collection:
+                              type: string
+                            db:
+                              type: string
+                          type: object
+                      required:
+                      - actions
+                      - resource
+                      type: object
+                    type: array
+                  role:
+                    description: The name of the role.
+                    type: string
+                  roles:
+                    description: An array of roles from which this role inherits privileges.
+                    items:
+                      description: Role is the database role this user should have
+                      properties:
+                        db:
+                          description: DB is the database the role can act on
+                          type: string
+                        name:
+                          description: Name is the name of the role
+                          type: string
+                      required:
+                      - db
+                      - name
+                      type: object
+                    type: array
+                required:
+                - db
+                - privileges
+                - role
                 type: object
               type: array
```

Comment:
The `type:array` field for roles here is initially set to object by the operator-sdk generate command. I'm not sure why, since roles is defined as an array field, but I had to manually change this to array to get it to work. Ideally I'd prefer not to introduce new peculiarities to the crd auto-generation, but I'm not sure why this happened.

Reply:
Fixed itself when I moved roles field to Spec.Security.Authentication.